WordPress/Classes/ViewRelated/Reader/ReaderHelpers.swift

```diff
@@ -218,26 +218,62 @@ struct ReaderPostMenuButtonTitles {
         }
     }
 
-
-    @objc open class func statsPropertiesForPost(_ post: ReaderPost, andValue value: AnyObject?, forKey key: String?) -> [AnyHashable: Any] {
-        var properties = [AnyHashable: Any]()
-        properties[WPAppAnalyticsKeyBlogID] = post.siteID
-        properties[WPAppAnalyticsKeyPostID] = post.postID
-        properties[WPAppAnalyticsKeyIsJetpack] = post.isJetpack
-        properties[WPAppAnalyticsKeyIsFollowing] = post.isFollowing
-        if let feedID = post.feedID, let feedItemID = post.feedItemID {
-            properties[WPAppAnalyticsKeyFeedID] = feedID
-            properties[WPAppAnalyticsKeyFeedItemID] = feedItemID
+    @objc open class func statsProperties(for post: AnyObject, mergeWith existing: [String: Any]? = nil) -> [String: Any] {
+        var dict: [String: Any] = [:]
+
+        var blogID = 0
+        var postID = 0
+        var feedID = 0
+        var feedItemID = 0
+        var isFollowing = false
+        var isJetpack = false
+
+        if let post = post as? ReaderPost {
+            if let temp = post.siteID {
+                blogID = Int(truncating: temp)
+            }
+            if let temp = post.postID {
+                postID = Int(truncating: temp)
+            }
+            if let temp = post.feedID {
+                feedID = Int(truncating: temp)
+            }
+            if let temp = post.feedItemID {
+                feedItemID = Int(truncating: temp)
+            }
+            isFollowing = post.isFollowing
+            isJetpack = post.isJetpack
+        } else if let post = post as? ReaderSiteTopic {
+            blogID = Int(truncating: post.siteID)
+            feedID = Int(truncating: post.feedID)
+            isFollowing = post.following
+            isJetpack = post.isJetpack
+        } else {
+            return existing ?? [:]
         }
 
-        if let value = value, let key = key {
-            properties[key] = value
+        if blogID != 0 {
```

Comment:
I discussed about that briefly with @ParaskP7 and we had prior discussions in which we agreed to add the properties even if they're `0`.

The reason is that having this field makes calculations easier, so you probably don't need to check for the presence or absence of this value and just include it. :)